### PR TITLE
[PSM Interop] Cover TestConfig version parsing with unit tests

### PR DESCRIPTION
**Pull request by sergiitk**
This is to make sure upgrading packaging module won't break our logic on version-based version skipping.
This also fixes a small issue with `dev-` prefix - it should only be allowed on the left side of the comparison.

Context: packaging module needs to be upgraded to be compatible with `blackd`.

### CODE REVIEW

**Comment by sergiitk** [comment]
cc @gnossen - At the moment this is to unblock upgrading `packaging` module. I don't know how difficult would it be to add a GitHub check for this. We may need to handle it in a different PR.

**Comment by sergiitk** [comment]
I'm working on a GitHub action [in my fork](https://github.com/sergiitk/grpc/pull/46), will make a separate PR once this is merged and my tests are done.

**Comment by sergiitk** [comment]
- [ ] [grpc/java/v1.57.x/branch/xds_k8s_lb](https://source.cloud.google.com/results/invocations/6a0f181b-1c52-4cdf-8d36-426f6cf93d63)
- [ ] [grpc/java/v1.57.x/branch/psm-security](https://source.cloud.google.com/results/invocations/0478bb4c-9d40-4683-bd48-fd16c5f865a8)
- [ ] [grpc/java/v1.57.x/branch/xds_url_map](https://source.cloud.google.com/results/invocations/d5f4edbf-4eab-4bbf-aac2-b0b90244aeee)

File: tools/run_tests/xds_k8s_test_driver/tests/unit/helpers/skips_test.py
Comment (XuanWang-Amos) [inline tools/run_tests/xds_k8s_test_driver/tests/unit/helpers/skips_test.py:75]: Nice use of parameters 👍

**Review by XuanWang-Amos — APPROVED** [review]

**Review by gnossen — APPROVED** [review]
Love it. Super thorough

**Pull request merged**
